### PR TITLE
Create Team Page

### LINKED ISSUE #504: Create team
Users who have signed in and integrated the Teams application via Connect Teams #501 are able to Create Team from the Teams index page.

Create Team enables the addition of all existing information of a team, including:
- Add team name, hashtag, description, location

### PR DESCRIPTION
**Pull request by kamicut**
* Closes #504 
* Adds a create team form page at `scoreboard/create-team`
* Creates a details form that can be reused by edit teams

### CODE REVIEW

**Comment by guidorice** [comment]
@kamicut  I am attempting to connect Scoreboard w/ osm-teams running locally, but getting a bit confused about updating my `.env` file. I am noticing that the Scoreboard Readme, when compared the `.env.sample` file it differs quite a lot. (See my notes about the varying set of environment variables in #476) The Readme and the .env.sample should be 100% consistent when it comes to environment variables.

Also I think the Scoreboard Readme should get a new section describing what steps are needed to setup Scoreboard as an osm-teams oauth app. Specifically what is the Callback URL for Scoreboard? i.e. what to enter into placeholder `https://mayapp/callback` in the OAuth2 settings form in osm-teams.

cc @LanesGood

File: components/teams/LocationInput.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import { useState } from 'react'
+import { Map, TileLayer, CircleMarker } from 'react-leaflet'
+import { reverse } from 'ramda'
+
+export default function LocationInput ({ value, onChange }) {
+  const [ zoom, setZoom ] = useState(1)
+  let newCenter = reverse(JSON.parse(value).coordinates)
```
Comment (guidorice) [inline components/teams/LocationInput.js]: Consider using `const` instead of `let`.

File: components/teams/LocationInput.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+  return (
+    <Map style={{ height: '300px' }}
+      center={newCenter}
+      zoom={zoom}
+      onViewportChanged={({ center, zoom }) => {
+        setZoom(zoom)
+        let toGeojson = `{
```
Comment (guidorice) [inline components/teams/LocationInput.js]: Consider using `const` instead of `let`.

File: components/teams/TeamDetailsForm.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
+import { useForm, Controller } from 'react-hook-form'
+import dynamic from 'next/dynamic'
+
+const LocationInput = dynamic(() => import('./LocationInput'), { ssr: false })
```
Comment (guidorice) [inline components/teams/TeamDetailsForm.js]: I am seeing a parsing error from vscode here and also in CampaignCard.js

> Parsing error: Unexpected token import

Are you seeing the same @kamicut? It seems the syntax is correct for Next.js and ES2020, but we may need to update either `.eslintrc.json` or open an issue on devseed-standard.

File: components/teams/TeamDetailsForm.js [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+export default function TeamDetailsForm ({ onSubmit, details }) {
```
Comment (guidorice) [inline components/teams/TeamDetailsForm.js]: Consider adding a test for TeamDetailsForm in test-ui modules.

File: components/teams/LocationInput.js [hunk 5]
```diff
@@ -0,0 +1,28 @@
+import { useState } from 'react'
+import { Map, TileLayer, CircleMarker } from 'react-leaflet'
+import { reverse } from 'ramda'
+
+export default function LocationInput ({ value, onChange }) {
```
Comment (guidorice) [inline components/teams/LocationInput.js]: Consider adding a test for LocationInput in test-ui modules.

File: pages/create-team.js [hunk 6]
```diff
@@ -0,0 +1,51 @@
+import React from 'react'
+import { connect } from 'unistore/react'
+import { actions } from '../lib/store'
+import TeamDetailsForm from '../components/teams/TeamDetailsForm'
+import Router from '../lib/router'
+
+function CreateTeam (props) {
```
Comment (guidorice) [inline pages/create-team.js]: Consider adding a test for CreateTeam in test-ui modules.

File: pages/create-team.js [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+  const handleSubmit = async (data) => {
+    try {
+      await props.createTeam(data)
+      Router.push('/teams')
+    } catch (e) {
+      console.error(e)
```
Comment (guidorice) [inline pages/create-team.js]: Consider removing this try/catch block. Unhandled exceptions should get logged to console anyways.

At first it looks like this block could fail silently for the user if there was an exception. But I believe `createTeam` prop is coming from `lib/store/actions/teams.js`, which has state management/notification and logging of the exceptions in it's `createTeam()`. If that is indeed case, then the try/catch block here is not doing anything and could be removed entirely.

**Comment by guidorice** [comment]
re: the issues with the Readme, the `.env.sample`, and getting  osm-teams running locally, @kamicut got me all setup and running. I am going to put that knowledge into #476 which is assigned to me.

File: components/teams/TeamDetailsForm.js (hunk 3 above)
Comment (kamicut) [reply components/teams/TeamDetailsForm.js]: @guidorice I'm not using a different eslint, are you getting an error when you run `yarn lint` ?

File: pages/create-team.js (hunk 7 above)
Comment (kamicut) [reply pages/create-team.js]: I think the idea here was to make sure that a Router push doesn't happen when there is an error (copied pattern from another part of the codebase). I'm going to take a look at the store and make sure that an exception doesn't still trigger the Router code.

Comment (guidorice) [reply pages/create-team.js]: Line 11 should not be reached at all if there is an exception from line 10- regardless of the try/catch block being there.

File: components/teams/TeamDetailsForm.js (hunk 3 above)
Comment (guidorice) [reply components/teams/TeamDetailsForm.js]: @kamicut ah good point to check: `yarn lint` reports no errors, but in my vscode editor, it's showing errors like:

Screen Shot 2020-02-26 at 2 26 07 PM

Apparently vscode think's it's running eslint but doing it differently than the devseed-standard script. Are you not seeing the same problem?

**Comment by kamicut** [comment]
@guidorice I've added a test for CreateTeam that loads all the children component except the map (because it requires extensive mocking) and makes sure they can be mount onto a page. I think writing tests for the sub components would have to have us rethink our entire testing approach for all components since we only test containers right now.

File: pages/create-team.js (hunk 7 above)
Comment (guidorice) [reply pages/create-team.js]: @kamicut can you respond or edit on this one?

**Comment by guidorice** [comment]
@kamicut re: component unit testing, I agree completely about mainly just testing containers. I think what you did in the new test 'Create Team renders without crashing' is sufficient. 

In my experience, even if we were using [Enzyme](https://enzymejs.github.io/enzyme/) for React tests it still only works very will with shallow mounted components, and deeply nested components are much more difficult.

Incidentally, `enzyme` is in in package.json but I don't see it being imported anywhere. Should we remove that dependency?

File: pages/create-team.js (hunk 7 above)
Comment (kamicut) [reply pages/create-team.js]: I removed the try catch @guidorice

**Pull request merged**